src/main/java/com/puppycrawl/tools/checkstyle/api/AuditEvent.java

```diff
@@ -74,11 +71,20 @@ public AuditEvent(Object src, String fileName) {
      * @param violation the actual violation
      */
     public AuditEvent(Object src, String fileName, Violation violation) {
-        super(src);
+        source = src;
```

Comment:
We should continue to throw an exception here for a null source, as we did previously  (via `EventObject`).

Reply:
Done.